### ISSUE TITLE
Cannot relates to fobi.contrib.plugins.form_handlers.db_store.SavedFormDataEntry

### ISSUE THREAD

**Issue by thomasWajs** [barseghyanartur/django-fobi#11]
I upgraded from version 0.4.19 to the latest 0.4.35

Following this upgrade, a model field cannot have a foreign key to the model SavedFormDataEntry from the db_store plugins anymore:

```python
from django.db import models
class Meh(models.Model):
    form_content = models.ForeignKey('SavedFormDataEntry')
```

```
$ python manage.py  makemigrations
ERRORS:
meh.TestFobi.form_content: (fields.E300) Field defines a relation with model 'SavedFormDataEntry', which is either not installed, or is abstract.
```

Another approach :

```python
from django.db import models
from fobi.contrib.plugins.form_handlers.db_store.models import SavedFormDataEntry
class TestFobi(models.Model):
    form_content = models.ForeignKey(SavedFormDataEntry)
```
```
$ python manage.py  makemigrations
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "d:\Programmation\Virtualenvs\test_fobi\lib\site-packages\django\core\management\__init__.py", line 385, in execute_from_command_line
    utility.execute()
  File "d:\Programmation\Virtualenvs\test_fobi\lib\site-packages\django\core\management\__init__.py", line 377, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "d:\Programmation\Virtualenvs\test_fobi\lib\site-packages\django\core\management\base.py", line 288, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "d:\Programmation\Virtualenvs\test_fobi\lib\site-packages\django\core\management\base.py", line 338, in execute
    output = self.handle(*args, **options)
  File "d:\Programmation\Virtualenvs\test_fobi\lib\site-packages\django\core\management\commands\makemigrations.py", line 111, in handle
    convert_apps=app_labels or None,
  File "d:\Programmation\Virtualenvs\test_fobi\lib\site-packages\django\db\migrations\autodetector.py", line 42, in changes
    changes = self._detect_changes(convert_apps, graph)
  File "d:\Programmation\Virtualenvs\test_fobi\lib\site-packages\django\db\migrations\autodetector.py", line 110, in _detect_changes
    self.new_apps = self.to_state.render()
  File "d:\Programmation\Virtualenvs\test_fobi\lib\site-packages\django\db\migrations\state.py", line 94, in render
    raise ValueError(msg.format(field=operations[0][1], model=lookup_model))
ValueError: Lookup failed for model referenced by field meh.TestFobi.form_content: meh.fobi.contrib.plugins.form_handlers.db_store.SavedFormDataEntry
```

I can confirm that this used to work on 0.4.19.
Thanks for any hint.

**Comment by barseghyanartur** [comment]
@thomasWajs 

I think it has something to do with this:

https://github.com/barseghyanartur/django-fobi/blob/master/src/fobi/contrib/plugins/form_handlers/db_store/__init__.py#L7

https://github.com/barseghyanartur/django-fobi/blob/0.4.19/src/fobi/contrib/plugins/form_handlers/db_store/__init__.py#L7

You should check what `app_label` does you content type table (in database have). And make the necessary changes. It should not be just one string anyway, when linking to models.

It should rather be:

    class Meh(models.Model):
        form_content = models.ForeignKey('db_store.SavedFormDataEntry')

Replace the "db_store.SavedFormDataEntry" with whatever value you have for the "app_label"."model" in the Django's content type table (django_content_type).

You're likely using Django 1.7, right? Can it be that Django had referenced the "db_store" plugin twice in the "django_content_type" table: once as "db_store"."savedformdataentry" and once as "fobi.contrib.plugins.form_handlers.db_store"."savedformdataentry"? Which apps do you have listed in `INSTALLED_APPS`?

It can be that instead of "db_store_savedformdataentry" table you have "fobi.contrib.plugins.form_handlers.db_store_savedformdataentry". In that case a simple rename would do the trick.

**Comment by thomasWajs** [comment]
Just to be 100% clear : I did nothing else on my existing project, beyond upgrading the version, and it worked like a charm in 0.4.19

I did a test on a fresh django project, using django 1.7 and python 2.7, with settings that looks like:
```fobi
INSTALLED_APPS = (
    'fobi',
    'fobi.contrib.plugins.form_handlers.db_store',
    'meh',
)
```
I did tried the ForeignKey('db_store.SavedFormDataEntry'), but no luck, same error.
Also tried ForeignKey('fobi.contrib.plugins.form_handlers.db_store.SavedFormDataEntry'), and still no luck.

The app_label with the fresh installation in the content type table is *fobi.contrib.plugins.form_handlers.db_store*.
In my old install, the app_label was *db_store*.

Obviously, there's something that changed on the app_label. However, renaminh in any way won't fix the issue.

**Comment by thomasWajs** [comment]
I just tried to use this AppConfig, and it fixed the issue 
```python
class Config(AppConfig):
        name  = 'fobi.contrib.plugins.form_handlers.db_store'
        label = 'db_store'
```

**Comment by barseghyanartur** [comment]
@thomasWajs 

To be honest, I'm a little bit confused. The `Django documentation <https://docs.djangoproject.com/en/1.7/ref/applications/#django.apps.AppConfig.label>`_ clearly says that labels should be unique too.

I'll need some time to spend on investigating it further. Thanks for bringing this up!

**Comment by thomasWajs** [comment]
Well, I was kinda surprised the first time I saw the AppConfig in fobi, because of the approach
```
name = label = 'path.to.module'
```
The documentation state that the label should be a valid Python identifier, which is not the case when you put dots inside (only alphanumerics and underscore are ok). It also says that it default to the last component of name, which seems like a sensible default.

However, you use a lot of applications inside fobi, so you may run into conflicts with thirs-party apps by using this default.
Maybe the best choice would be to label explicitly all AppConfigs, using a "fobi_" prefix instead of the full dotted path:
```python
class Config(AppConfig):
        name  = 'fobi.contrib.plugins.form_handlers.db_store'
        label = 'fobi_db_store'
```

This would lead to a backward-incompatibility as all db tables will change name, as well as content types.

**Comment by barseghyanartur** [comment]
@thomasWajs 

Ah, great for finding this out! I completely misunderstood that part. Some things may break indeed, but not if I provide the `db_table` argument in the model. I'll be replacing dots in the `name` with underscores. Most of the `fobi` models are in core (having`fobi` as a label). As for the contrib apps, there are just a few that have them. I think I'll release the 0.5 tonight with fixes and shift the planned development up by one minor version.

Thanks again.

**Comment by barseghyanartur** [comment]
@thomasWajs 

Fixed in django-fobi==0.5. There are no automated migrations provided. Table names and references should be updated manually.
